cli/data_loader.py

```diff
@@ -0,0 +1,120 @@
+#
+# Copyright (c) 2019, Neptune Labs Sp. z o.o.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from __future__ import print_function
+
+import os
+import click
+import mlflow
+import path as path_utils
+
+class DataLoader(object):
+
+    MLFLOW_EXPERIMENT_ID_PROPERTY = "mlflow/experiment/id"
+    MLFLOW_EXPERIMENT_NAME_PROPERTY = "mlflow/experiment/name"
+    MLFLOW_RUN_ID_PROPERTY = "mlflow/run/uuid"
+    MLFLOW_RUN_NAME_PROPERTY = "mlflow/run/name"
+
+    def __init__(self, project, path):
+        self._project = project
+        self._path = path
+
+    def run(self):
+        with path_utils.Path(self._path):
+            mlflow_client = mlflow.tracking.MlflowClient()
+            experiments = mlflow_client.list_experiments()
+
+            for experiment in experiments:
+                run_infos = mlflow_client.list_run_infos(experiment_id=experiment.experiment_id)
+                existing_experiments = self._project.get_experiments(tag=experiment.name.lower())
+                existing_run_uuids = set([
+                    str(e.get_properties().get(DataLoader.MLFLOW_RUN_ID_PROPERTY)) for e in existing_experiments
+                ])
+
+                for run_info in run_infos:
+                    run_qualified_name = self._get_run_qualified_name(experiment, run_info)
+                    if run_info.run_uuid not in existing_run_uuids:
+                        click.echo("Loading run {}".format(run_qualified_name))
+                        run = mlflow_client.get_run(run_info.run_uuid)
+                        exp_uuid = self._create_neptune_experiment(experiment, run)
```

Comment:
The state of the experiment will be `running` forever?

Reply:
Experiment is created via `with` so context manager will set it to `succeeded`.